src/OrbitBase/include/OrbitBase/Typedef.h

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_TYPEDEF_H_
+#define ORBIT_BASE_TYPEDEF_H_
+
+#include <functional>
+#include <type_traits>
+#include <utility>
+
+namespace orbit_base {
+
+template <typename T, typename U>
+using EnableIfUConvertibleToT = std::enable_if_t<std::is_convertible_v<U, T>>;
+
+template <typename T, typename U>
+using EnableIfUIsNotConvertibleToT = std::enable_if_t<!std::is_convertible_v<U, T>>;
+
+// Strong typedef
```

Comment:
A little more elaborate explanation would be great. :-) Maybe even a small example on how to use that with `Apply` for the most basic use-case.